### ISSUE TITLE
`pgpainless-cli verify  gives error "does not contain an OpenPGP certificate" when signature does not contain a signature

### ISSUE THREAD

**Issue by dkg** [vanitasvitae/pgpainless#330]
i think the error message is wrong here -- it should be checking whether the first positional argument to `pgpainless-cli verify` is a *signature*, not whether it is a *certificate*.

here's a log of the failure:

```
0 dkg@alice:$ pgpainless-cli generate-key foo > foo.key
0 dkg@alice:$ echo test > test.txt
0 dkg@alice:$ pgpainless-cli extract-cert < foo.key > foo.crt
0 dkg@alice:$ echo not a sig > not-a-sig.txt
0 dkg@alice:$ pgpainless-cli verify not-a-sig.txt foo.crt < test.txt 
Input 'not-a-sig.txt' does not contain an OpenPGP certificate.
sop.exception.SOPGPException$BadData: Input 'not-a-sig.txt' does not contain an OpenPGP certificate.
	at sop.cli.picocli.commands.VerifyCmd.run(VerifyCmd.java:70)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at sop.cli.picocli.SopCLI.execute(SopCLI.java:84)
	at org.pgpainless.cli.PGPainlessCLI.execute(PGPainlessCLI.java:29)
	at org.pgpainless.cli.PGPainlessCLI.main(PGPainlessCLI.java:22)
Caused by: sop.exception.SOPGPException$BadData: java.io.IOException: unknown object in stream: 46
	at org.pgpainless.sop.DetachedVerifyImpl.cert(DetachedVerifyImpl.java:47)
	at org.pgpainless.sop.DetachedVerifyImpl.cert(DetachedVerifyImpl.java:25)
	at sop.cli.picocli.commands.VerifyCmd.run(VerifyCmd.java:65)
	... 11 more
Caused by: java.io.IOException: unknown object in stream: 46
	at org.bouncycastle.openpgp.PGPObjectFactory.nextObject(Unknown Source)
	at org.pgpainless.key.parsing.KeyRingReader.readPublicKeyRingCollection(KeyRingReader.java:246)
	at org.pgpainless.key.parsing.KeyRingReader.readPublicKeyRingCollection(KeyRingReader.java:222)
	at org.pgpainless.key.parsing.KeyRingReader.publicKeyRingCollection(KeyRingReader.java:86)
	at org.pgpainless.sop.DetachedVerifyImpl.cert(DetachedVerifyImpl.java:45)
	... 13 more
41 dkg@alice:$ 
```

**Comment by dkg** [comment]
forgot to add:

```
0 dkg@alice:$ pgpainless-cli version --extended
PGPainless-SOP 1.3.7
https://codeberg.org/PGPainless/pgpainless/src/branch/master/pgpainless-sop

Implementation of the Stateless OpenPGP Protocol Version 04
https://datatracker.ietf.org/doc/html/draft-dkg-openpgp-stateless-cli-04

Based on pgpainless-core 1.3.7
https://pgpainless.org

Using Bouncy Castle 1.71
https://www.bouncycastle.org/java.html
0 dkg@alice:$ 
```

**Comment by vanitasvitae** [comment]
That's indeed an issue, thanks for the report!

**Comment by vanitasvitae** [comment]
Heh, apparently I already had fitting error messages, but somehow in one faulty commit I changed the arity of arguments, causing the signature to be included in the list of certificates.

**Comment by vanitasvitae** [comment]
I just released `sop-java` in version `4.0.2` and `pgpainless-cli` in version `1.3.9` to address those issues.
